src/Models/incompressible_model.jl

```diff
@@ -80,34 +82,45 @@ function IncompressibleModel(;
           surface_waves = nothing,
                 forcing = NamedTuple(),
                 closure = IsotropicDiffusivity(float_type, ν=ν₀, κ=κ₀),
-    boundary_conditions = (u=UVelocityBoundaryConditions(grid),
-                           v=VVelocityBoundaryConditions(grid),
-                           w=WVelocityBoundaryConditions(grid)),
+    boundary_conditions = NamedTuple(),
                 tracers = (:T, :S),
             timestepper = :QuasiAdamsBashforth2,
-             velocities = VelocityFields(architecture, grid, boundary_conditions),
-              pressures = PressureFields(architecture, grid, boundary_conditions),
-          diffusivities = DiffusivityFields(architecture, grid, tracernames(tracers), boundary_conditions, closure),
-        pressure_solver = PressureSolver(architecture, grid, PressureBoundaryConditions(grid))
+             velocities = nothing,
+              pressures = nothing,
+          diffusivities = nothing,
+        pressure_solver = nothing,
     )
 
     if architecture == GPU() && !has_cuda()
          throw(ArgumentError("Cannot create a GPU model. No CUDA-enabled GPU was detected!"))
     end
 
+    tracers = tupleit(tracers) # supports tracers=:c keyword argument (for example)
     validate_buoyancy(buoyancy, tracernames(tracers))
 
-    # Regularize forcing and closure for model tracer and velocity fields.
-    forcing = model_forcing(tracernames(tracers); forcing...)
-    closure = with_tracers(tracernames(tracers), closure)
+    # Adjust halos when the advection scheme or turbulence closure requires it.
+    # Note that halos are isotropic by default; however we respect user-input here
+    # by adjusting each (x, y, z) halo individually.
+    Hx, Hy, Hz = inflate_halo_size(grid.Hx, grid.Hy, grid.Hz, advection, closure)
+    grid = with_halo((Hx, Hy, Hz), grid)
```

Comment:
Should we print an `@info` that halo sizes were inflated or adjusted? In case users pass a grid with funny halo sizes so they don't get surprised?

I guess most users will just pick an advection scheme and won't think about halos so the `@info` message will show up pretty frequently.

But maybe more package->user feedback like this is good and helps stave off frustration due to long compile times?

Reply:
I agree with this idea. But what if we put a message in the model constructor that says something like

```
INFO: Constructing model on RegularCartesianGrid <<info about grid that includes halos>>
```

Should we open an issue about this and develop a coherent plan towards informative messages?

Reply:
New issue in #1013

Reply:
Makes sense, nice issue!